File: pkg/java/src/test/java/dev/openfga/language/util/JsonValidationTestCase.java
package dev.openfga.language.util;

import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

import dev.openfga.language.errors.ModelValidationSingleError;

public class JsonValidationTestCase {
    @JsonProperty("name")
    @JsonInclude(JsonInclude.Include.USE_DEFAULTS)
    private String name;

    @JsonProperty("json")
    @JsonInclude(JsonInclude.Include.USE_DEFAULTS)
    private String json;

    @JsonProperty("expected_errors")
    @JsonInclude(JsonInclude.Include.USE_DEFAULTS)
    private List<ModelValidationSingleError> expectedErrors = new ArrayList<>();



    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getJson() {
        return json;
    }

    public void setJson(String json) {
        this.json = json;
    }

    public List<ModelValidationSingleError> getExpectedErrors() {
        return expectedErrors;
    }

    public void setExpectedErrors(List<ModelValidationSingleError> expectedErrors) {
        this.expectedErrors = expectedErrors;
    }
}

File: pkg/csharp/test/OpenFga.Language.Tests/util/JsonValidationTestCase.cs
using OpenFga.Language.Validation;
using System.Text.Json.Serialization;
using OpenFga.Language.Errors;
using SharpYaml.Serialization;

namespace OpenFga.Language.Tests.util;

public class JsonValidationTestCase
{
    [YamlMember("name")]
    public string Name { get; set; } = string.Empty;

    [YamlMember("json")]
    public string Json { get; set; } = string.Empty;

    [YamlMember("expected_errors")]
    public List<ModelValidationSingleError>? ExpectedErrors { get; set; } = new();
}
